rest-helix/src/main/java/com/github/twitch4j/helix/interceptor/TwitchHelixHttpClient.java

```diff
@@ -99,10 +100,47 @@ private Response delegatedExecute(Request request, Request.Options options) thro
             return executeAgainstBucket(clipBucket, () -> client.execute(request, options));
         }
 
+        // Extensions API: sendExtensionChatMessage has a stricter per-channel bucket
+        if (request.httpMethod() == Request.HttpMethod.POST && templatePath.endsWith("/extensions/chat")) {
+            // Obtain the bucket key
+            String clientId = request.headers().getOrDefault(CLIENT_HEADER, Collections.emptyList()).iterator().next();
+            String channelId = request.requestTemplate().queries().getOrDefault("broadcaster_id", Collections.emptyList()).iterator().next();
+
+            // Conform to endpoint-specific bucket
+            Bucket chatBucket = rateLimitTracker.getExtensionChatBucket(clientId, channelId);
+            return executeAgainstBucket(chatBucket, () -> client.execute(request, options));
+        }
+
+        // Extensions API: sendExtensionPubSubMessage has a stricter bucket depending on the target
+        if (request.httpMethod() == Request.HttpMethod.POST && templatePath.endsWith("/extensions/pubsub")) {
+            // Obtain the bucket key
+            String clientId = request.headers().getOrDefault(CLIENT_HEADER, Collections.emptyList()).iterator().next();
+            String target = getExtensionPubSubTarget(request.body());
+
+            // Conform to endpoint-specific bucket
+            Bucket pubSubBucket = rateLimitTracker.getExtensionPubSubBucket(clientId, target);
+            return executeAgainstBucket(pubSubBucket, () -> client.execute(request, options));
+        }
+
         // no endpoint-specific rate limiting was needed; simply perform network request now
         return client.execute(request, options);
     }
 
+    static String getExtensionPubSubTarget(byte[] body) {
```

Comment:
does this belong in a different class

Reply:
a possible alternative approach: add the target as an extra query param (that we make up) & hope twitch ignores it

Reply:
Would putting it into a header field and then taking it out of there be a option (easier to access than get-params)?
We could also use `request.headers().remove("KEY");` to get rid of the header before the request is sent in here.

Reply:
Implemented the header approach, but didn't remove the header because of the bucket update post-request